### ISSUE TITLE
Documentation: on vim, specify "set nocompatible" when using the -u option

### ISSUE THREAD

**Issue by nerfologist** [k-takata/minpac#67]
## TLDR
In vim (not neovim) `set nocompatible` seems to be necessary for minpac to init and work properly. This line might usually come from a system vimrc; but that file is not read when loading a custom local vimrc with the `-u` flag, and this might puzzle some user.

I would add a documentation notice to explicitly declare this.

## Long explaination
Hello,
when loading brew macvim, version
```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Aug 31 2018 10:52:37)
macOS version
Included patches: 1-280
Compiled by Homebrew
Huge version with MacVim GUI.  Features included (+) or not (-):
...
```
I had to load a custom `vimrc` file for some tests.

I initially made a custom `vimrc`, same as the minimal minpac example from README.md
```
" vimrc
packadd minpac

call minpac#init()

" minpac must have {'type': 'opt'} so that it can be loaded with `packadd`.
call minpac#add('k-takata/minpac', {'type': 'opt'})

" Add other plugins here.
call minpac#add('vim-jp/syntax-vim-ex')
...

" Load the plugins right now. (optional)
"packloadall
```
and I was running macvim with this custom file only, using
```
/usr/local/bin/vim -u ./vimrc
```

Unfortunately, `minpac#init()` was not working: 
```
Error detected while processing function minpac#init:
line    1:
E116: Invalid arguments for function extend(copy(get(a:000, 0, {})),
E15: Invalid expression: extend(copy(get(a:000, 0, {})),
Press ENTER or type command to continue
```

I found out this is due to the following line being missing from my custom configuration:

```
set nocompatible
```
...which in my Macvim/brew version is usually found in `/usr/local/Cellar/macvim/8.1-151/MacVim.app/Contents/Resources/vim/vimrc` (system vimrc). This is part of the usual vim configuration file hierarchy when loading vim normally.

Unfortunately, when loading a custom vimrc file using the `-u` flag, that file is NOT read.

I think it would be a good idea to specify in the documentation that this plugin requires `set nocompatible` to init.

**Comment by k-takata** [comment]
It is the documented behavior of `-u`. See `:help -u`:

```
		Using the "-u" argument with another argument than DEFAULTS
		has the side effect that the 'compatible' option will be on by
		default.  This can have unexpected effects.  See
		|'compatible'|.
```

If you want to use `-u`, you should also specify `-N`: `vim -N -u vimrc`.
It's not specific to minpac.

**Comment by nerfologist** [comment]
Hello,
thanks for your answer, I know this is documented, and the issue is not about the specific `-u` behavior, but rather on clarifying that minpac requires `nocompatible`, which may not be evident from the sample `.vimrc` (although `set nocompatible` is included in the following on demand `.vimrc`).

**Comment by k-takata** [comment]
Updated the document in `devel` branch.

**Comment by nerfologist** [comment]
Thanks for your changes, closing this now 👍
